src/utils.ts

```
@@ -47,3 +49,45 @@ export async function getAppDataSchema(version: string): Promise<AnyAppDataDocVe
     throw new Error(`AppData version ${version} doesn't exist`)
   }
 }
+
+export type ValidationResult = {
+  success: boolean
+  errors?: string
+}
+
+let ajv: Ajv
+
+export async function validateAppDataDoc(appDataDoc: AnyAppDataDocVersion): Promise<ValidationResult> {
+  const version = appDataDoc.version
+
+  if (!ajv) {
+    ajv = new Ajv()
+  }
+
+  let validator = ajv.getSchema(version)
+
+  if (!validator) {
+    let schema
+    try {
+      schema = await getAppDataSchema(version)
+    } catch (e) {
+      if (e instanceof Error) {
+        return {
+          success: false,
+          errors: e.message,
+        }
+      } else {
+        throw e
+      }
+    }
+
+    ajv.addSchema(schema, version)
```

why not add this into the `try` block and not use the external `let`? or am i reading the brackets wrong

---

You read it right.
I wanted to make sure the error can only come form `getAppDataSchema` rather than swallow any errors adding or getting the schema to `ajv`